anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -322,6 +322,55 @@ def image_content_to_grype_sbom(self, record: Dict):
         return artifact
 
 
+class JavaMapper(CPEMapper):
+
+    @staticmethod
+    def _image_content_to_grype_metadata(metadata: Optional[dict]) -> dict:
+        result = None
+
+        if not metadata:
+            return result
+
+        result = copy.deepcopy(metadata)
+        pom_properties = result.get("pom.properties")
+
+        if not pom_properties:
+            return result
+        elif isinstance(pom_properties, dict):
+            result["pomProperties"] = pom_properties
+        else:
+            # If the pom.properties block is a newline-delimited-list instead of a dict, transform it,
+            # accounting for leading and trailing `\n` delimiters
+            transformed_pom_properties = {}
+
+            split_pom_properties = [
+                item for item in pom_properties.split("\n") if item != ""
+            ]
+            for item in split_pom_properties:
+                # Expected input here is like "groupId=org.yaml" or "version=1.23"
+                split_item = item.split("=", 2)
+                if len(split_item) < 2:
+                    log.warn(
+                        "Skipping malformed java package metadata pom property: %s ", item
+                    )
+                    continue
+                transformed_pom_properties[split_item[0]] = split_item[1]
+            result["pomProperties"] = transformed_pom_properties
+
+        # grype expects this field to be called "pomProperties", not "pom.properties"
+        del result["pom.properties"]
+
+        return result
+
+    def image_content_to_grype_sbom(self, record: Dict):
+        artifact = super().image_content_to_grype_sbom(record)
+
+        grype_metadata = self._image_content_to_grype_metadata(record.get("metadata"))
+        if grype_metadata:
+            artifact["metadataType"] = "JavaMetadata"
+            artifact["metadata"] = grype_metadata
+
+
```

Comment:
@dspalmer99 Found the issue, return statement is missing. I made the same mistake with kb mapper the other day. Will add it along with a unit test